### PR TITLE
Allow dashes in GUI pvs

### PR DESCRIPTION
**Pull request by Tom-Willemsen**
### Description of work

Allows dashes in PV names.

### To test

https://github.com/ISISComputingGroup/IBEX/issues/2580

### Acceptance criteria

Client allows you to use dashes in PV names.

Check:
- Instrument switcher, custom instrument dialog
- Blocks dialog
- Synoptics dialog
- That you still can't have a "-" in a block name

---

#### Code Review

- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/GUI-Coding-Conventions)? Is it well structured with small focussed classes/methods/functions?
- [ ] Have no new checkstyle warnings been introduced? Check via Jenkins
- [ ] Are there unit tests in place? Are the unit tests small and test the a class in isolation?
- [ ] Are there automated [system tests](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/System-Testing-with-RCPTT) in place? Do they test a minimal set of functionality and leave the gui as close as possible to its original state?
- [ ] Has the [manual system tests spreadsheet](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Manual-system-tests) been updated?
    - Manual system tests for the functionality should be added if there are no automated tests
    - Manual system tests can be removed from the template if they are covered by suitable automated tests
- [x] Did any existing system test break as a result of the current changes? 
- [x] Have the changes been documented in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev). If so, do they describe the changes appropriately?
- [ ] If an OPI has been modified, does it conform to the [style guidelines](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/OPI-Creation)? There is a script called `check_opi_format.py` to help with this.

### Functional Tests

- [x] Do changes function as described? Add comments below that describe the tests performed.
- [x] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has developer documentation been updated if required?
- [ ] Have any new plugins been added to the appropriate feature.xml? You can check if this is needed by validating plugins as per method at  https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Common-Eclipse-Issues

### Final Steps
- [ ] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

### CODE REVIEW

File: base/uk.ac.stfc.isis.ibex.validators.tests/src/uk/ac/stfc/isis/ibex/validators/tests/PvValidatorTest.java [hunk 1]
```diff
@@ -65,6 +65,16 @@ public void GIVEN_pv_address_with_non_alpha_character_WHEN_validate_THEN_valid()
...
 
     @Test
+    public void GIVEN_pv_address_with_dash_WHEN_validate_THEN_valid() {
+        // Arrange
+        String testAddress = "va_:l-d.fi_:eld";
+        // Act
+        PvValidator addressValid = new PvValidator();
+        // Assert
+        assertTrue(addressValid.validatePvAddress(testAddress));
+    }
```
Comment (ThomasLohnert) [inline base/uk.ac.stfc.isis.ibex.validators.tests/src/uk/ac/stfc/isis/ibex/validators/tests/PvValidatorTest.java]: Add an equivalent test to InstrumentNameValidatorTest

File: base/uk.ac.stfc.isis.ibex.validators/src/uk/ac/stfc/isis/ibex/validators/PvValidator.java [hunk 2]
```diff
@@ -56,7 +56,7 @@ public Boolean validatePvAddress(String pvAddress) {
 
         if (pvAddress.isEmpty()) {
             setErrorMessage(ADDRESS_EMPTY);
-        } else if (!(pvAddress.matches("^[a-zA-Z0-9_:]+(\\.[a-zA-Z0-9_:]+)?$"))) {
+        } else if (!(pvAddress.matches("^[a-zA-Z0-9_\\-:]+(\\.[a-zA-Z0-9_:\\-]+)?$"))) {
```
Comment (ThomasLohnert) [inline base/uk.ac.stfc.isis.ibex.validators/src/uk/ac/stfc/isis/ibex/validators/PvValidator.java]: I noticed that this currently allows PVs with multiple colons (e.g. `IN:DEMO::SOME:PV`). Would it make sense to fix this as part of this PR?

File: base/uk.ac.stfc.isis.ibex.validators/src/uk/ac/stfc/isis/ibex/validators/PvValidator.java [hunk 3]
```diff
@@ -29,7 +29,7 @@
 public class PvValidator {
 
     public static final String ADDRESS_FORMAT =
-            "PV Address invalid, use only letters, numbers, underscores and colons";
+            "PV Address invalid, use only letters, numbers, underscores, dashes and colons";
     public static final String ADDRESS_EMPTY = "PV Address invalid, must not be empty";
```
Comment (ThomasLohnert) [inline base/uk.ac.stfc.isis.ibex.validators/src/uk/ac/stfc/isis/ibex/validators/PvValidator.java]: Fix checkstyle warnings for these two variables

File: base/uk.ac.stfc.isis.ibex.validators/src/uk/ac/stfc/isis/ibex/validators/PvValidator.java (hunk 2 above)
Comment (Tom-Willemsen) [reply base/uk.ac.stfc.isis.ibex.validators/src/uk/ac/stfc/isis/ibex/validators/PvValidator.java]: I had a quick look at this but it's not completely trivial, you need to craft a regex that allows any number of colon-seperated strings but disallows consecutive colons. Definitely possible but I'd rather not mix it in with this change.

I've created ticket https://github.com/ISISComputingGroup/IBEX/issues/2644 to take a look at this.

**Pull request merged**